### PR TITLE
Making .on() fire for multiple delegated handlers on the same element and event.

### PR DESCRIPTION
**Pull request by barndawe**

### CODE REVIEW

File: docs/test.js [hunk 1]
```diff
@@ -2074,13 +2074,14 @@ describe(".on(event, fn)", function () {
         done();
       }
     });
-    u('.cb-wrapper').on('change', '.cb', function () {
+    base.find('.cb-wrapper').on('change', '.cb', function () {
```
Comment (franciscop) [inline docs/test.js:2077]: Why is this change needed?

File: src/plugins/on/test.js [hunk 2]
```diff
@@ -152,12 +152,13 @@ describe(".on(event, fn)", function () {
         done();
       }
     });
-    u('.cb-wrapper').on('change', '.cb', function () {
+    base.find('.cb-wrapper').on('change', '.cb', function () {
```
Comment (franciscop) [inline src/plugins/on/test.js:155]: Same here, why is this change needed?

**Comment by franciscop** [comment]
Hi @barndawe, thanks for the PR. Could you please provide a description on the PR? Also specially importantly please explain why the current tests need to change, since that seems like it'd be breaking backwards-compatibility.

File: src/plugins/on/test.js (hunk 2 above)
Comment (barndawe) [reply src/plugins/on/test.js:155]: I made this change to my test to better match the style of the existing tests. I noticed after the fact that none of the I other tests were directly using u().

File: docs/test.js (hunk 1 above)
Comment (barndawe) [reply docs/test.js:2077]: This must've propagated from on/test.js as part of the build.

**Comment by barndawe** [comment]
Hi @franciscop, apologies, I've never submitted a PR for anything before. 
The reasoning behind it is that I noticed if I attached more than one event handler via delegates then only one of them would fire. I tracked it down to the on() method changing the event's currentTarget in the delegation wrapper callback such that u(e.currentTarget).find(sel) won't work for subsequent handlers for the same event instance.

As for the tests I have only added one new one for my change, which I then tidied up slightly in a subsequent commit.

**Comment by franciscop** [comment]
True, I was reviewing only that commit for some reason and missed that. This looks great, thank you! I'll publish it shortly

**Comment by franciscop** [comment]
Published as version `3.2.3`, thanks!
